cds/symptom_report.go

```diff
@@ -0,0 +1,154 @@
+package cds
+
+import (
+	"net/http"
+	"sort"
+	"time"
+
+	"github.com/gin-gonic/gin"
+
+	"github.com/bitmark-inc/data-store/store"
+)
+
+func (cds *CDS) AddSymptomDailyReports() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		var body struct {
+			Reports []*store.SymptomDailyReport `json:"reports"`
+		}
+
+		if err := c.Bind(&body); err != nil {
+			c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+			return
+		}
+
+		if err := cds.dataStorePool.Community().AddSymptomDailyReports(c, body.Reports); err != nil {
+			c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+			return
+		}
+
+		c.JSON(http.StatusOK, gin.H{"result": "ok"})
+	}
+}
+
+type reportItemQueryParams struct {
+	Start string `form:"start"`
+	End   string `form:"end"`
+}
+
+type reportItem struct {
+	ID           string         `json:"id"`
+	Name         string         `json:"name"`
+	Value        *int           `json:"value"`
+	ChangeRate   *float64       `json:"change_rate"`
+	Distribution map[string]int `json:"distribution"`
+}
+
+func (cds *CDS) GetSymptomReportItems() gin.HandlerFunc {
+	return func(c *gin.Context) {
```

Comment:
We do not need a closure here. I would rather make `GetSymptomReportItems` as a `gin.HandlerFunc` directly.